extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -301,4 +351,114 @@ private void removeCookie(RoutingContext context, TenantConfigContext configCont
             }
         }
     }
+
+    private boolean isLogout(RoutingContext context, TenantConfigContext configContext) {
+        Optional<String> logoutPath = configContext.oidcConfig.logout.path;
+
+        if (logoutPath.isPresent()) {
+            return context.request().absoluteURI().equals(
+                    buildUri(context, logoutPath.get()));
+        }
+
+        return false;
+    }
+
+    private SecurityIdentity trySilentRefresh(TenantConfigContext configContext, String idToken, String refreshToken,
+            RoutingContext context, IdentityProviderManager identityProviderManager) {
+        OidcTenantConfig config = configContext.oidcConfig;
+
+        if (config.token.refreshTimeout.isPresent()) {
+            OAuth2AuthProviderImpl auth = ((OAuth2AuthProviderImpl) configContext.auth);
+            JWT jwt = auth.getJWT();
+            JsonObject tokenJson;
+
+            try {
+                tokenJson = jwt.decode(idToken);
+            } catch (Exception cause) {
+                throw new AuthenticationCompletionException(cause);
+            }
+
+            Long iat = tokenJson.getLong("iat");
+            long now = System.currentTimeMillis() / 1000;
+            Integer timeout = config.token.refreshTimeout.get();
+
+            if (now - iat >= timeout) {
+                String sub = tokenJson.getString("sub");
+
+                TokenEntry entry = refreshingTokens.computeIfAbsent(sub,
```

Comment:
Indeed, it makes it stateful. You may have noticed that the idea is to prevent requests from failing when they are sent from a same subject at the same time.

The point here is making requests consistent and don't fail due to multiple refresh token requests from a same subject. This is especially useful when you have refresh token rotation enabled at the IdP and after a refresh token request, you get also a new refresh token.

I understand the concerns you and @sberyozkin have about this part. In fact, my initial changes were just failing requests. But I do think this is useful and important to make things consistent.

So, if you *also* think we should remove this, I will not argue anymore and just remove it.

Reply:
The big problem with this is that it is not 'cloud native'. Requests may be going to different pods, so you can't be sure that this state is actually shared between requests.

Reply:
That is a good argument. Missed that, so that in the case of pods, pod-affinity is an anti-pattern.

I'm going to remove this part and just fail the request. Let's treat it as a corner case and have this behavior documented properly.